src/renderer/components/test-utils/get-application-builder.tsx

```diff
@@ -213,6 +218,35 @@ export const getApplicationBuilder = () => {
     },
   }));
 
+  mainDi.override(lensFetchInjectable, (di) => {
+    return async (pathnameAndQuery, init) => {
+      const handleLensRequest = di.inject(handleLensRequestInjectable);
+      const { Headers, Response } = di.inject(nodeFetchModuleInjectable);
+
+      const url = new URL(pathnameAndQuery, "https://127.0.0.1");
+      const req = httpMocks.createRequest({
+        method: (init?.method ?? "get").toUpperCase() as any,
+        url: url.pathname,
+        params: url.searchParams,
+        body: (init?.body ?? undefined) as any,
+        headers: new Headers(init?.headers) as any,
+      });
+      const duplex = new stream.Duplex();
+      const res = httpMocks.createResponse({
+        req,
+        writableStream: duplex,
+      });
+
+      await handleLensRequest.handle(req, res);
+
+      return new Response(duplex, {
+        headers: new Headers(res._getHeaders() as Record<string, string>),
+        status: res._getStatusCode(),
+        statusText: res._getStatusMessage(),
+      });
+    };
+  });
+
```

Comment:
Interesting fake! Thinking real hard about the ramifications of it (note to self).